src/convert/impls.rs

```diff
@@ -223,6 +224,20 @@ impl<T> FromWasmAbi for *const T {
     }
 }
 
+impl<T> OptionIntoWasmAbi for *const T {
+    #[inline]
+    fn none() -> u32 {
+        0
+    }
+}
+
+impl<T> OptionFromWasmAbi for *const T {
+    #[inline]
+    fn is_none(js: &u32) -> bool {
+        *js == 0
+    }
+}
```

Comment:
This doesn't work: since `*const T` describes itself as a `u32`, as far as the CLI's concerned an `Option<*const T>` is the same as an `Option<u32>`, and `Option<u32>` doesn't use this representation. It directly implements `IntoWasmAbi`, without using `OptionIntoWasmAbi`, and works by adding an extra `u32` that represents 'is this some?'.

If you wanted to, you could add a new descriptor for pointers that does work this way, but I think doing it the same way as `Option<u32>` is better anyway since it's perfectly valid for a `*const T` to be null, and so that shouldn't be treated as some sentinel value.

On the other hand, null isn't a valid value for `NonNull<T>`, and so doing it this way for that would be fine (but it'd need a new descriptor).